### DISCUSSION TITLE
Using alongside Vue.js

### DISCUSSION THREAD

**Discussion by james0r** [EvgeniyMukhamedjanov/liquid-ajax-cart#20] [Q&A] [answered]
Just curious if there are any gotchas using LAC alongside Vue.js. I've been using [https://themelab.uicrooks.com/](https://themelab.uicrooks.com/) as a starter and usually use LAC alongside Alpine.js which does direct DOM manipulation unlike Vue.js.

I'll be playing around with the two but just curious if there were any forewarnings. Thanks!

**Comment by EvgeniyMukhamedjanov** [comment, marked as answer]
Nothing special comes to my mind. BTW if you want to build the content of the cart on Vue then you can use the [`subscribeToCartStateUpdate`](https://liquid-ajax-cart.js.org/reference/subscribeToCartStateUpdate/) function that returns all the info about cart state and pass this object to the Vue component. In this case you don't need to use the the `data-ajax-cart-section`.
Otherwise keep in mind that if you want to use the `data-ajax-cart-section` attribute instead of Vue components for the cart  — it must be inside a Shopify section.

**Comment by james0r** [comment]
Okay awesome. i'll play around with it using my Shopify Vue starter. Also, might be cool to see liquid-ajax-cart as an Alpine.js plugin. Seems like they would pair really well. There's no problem using both alongside each other currently, but it could allow for a cleaner project and offer reactivity. CartJS used a pairing like this back in the day with rivets. Don't get me wrong this thing is awesome stand-alone and I love the energy you're putting into it. Maybe i'll play around with LAC as an Alpine plugin in the future.

**Reply by EvgeniyMukhamedjanov** [reply]
Yeah, I remember rivets :-)
But the problem is if you want to build your cart dynamically using rivets or Vue or any other reactivity stuff then you have to pass the JSON data that Shopify returns after any cart request. In CartJS you can check what the data is using `CartJS.cart` expession, in LAC it is the `getCartState()` method. If you check it out you can see that it is pretty limited data in comparison to what data you can have if you use liquid templates. As the example: once I needed to get line item product's tag on a CartJS project but it is not possible.

But if you want to use LAC with reactive things like rivets or Vue then just use the `subscribeToCartStateUpdate()` fn and pass the JSON data to rivets root or Vue root component.

I'll be happy to see Alpine plugin btw :-)
Keep me posted.

**Comment by james0r** [comment]
After looking at the potential benefits of an Alpine plugin, it seems just syncing a global Alpine prop using `subscribeToCartStateUpdate()` like you suggested makes way more sense.

I'm now importing LAC via the NPM package for easier upgrades. My Alpine store looks like this

```
import {
  subscribeToCartAjaxRequests,
  subscribeToCartStateUpdate,
  cartRequestGet,
} from 'liquid-ajax-cart'

export default {
  name: 'liquidAjaxCart',
  store() {
    return {
      init() {
        document.addEventListener('alpine:init', () => {
          let Alpine = window.Alpine

          subscribeToCartStateUpdate((state) => {
            Alpine.store('liquidAjaxCart', {
              state: state,
            })
          })

          cartRequestGet()
        })
      },
    }
  },
}
```

It seems I have to wrap `subscribeToCartStateUpdate()` in the `alpine:init` lifecycle hook or I get errors when my alpine directive expressions are evaluated before the state $store prop is initialized.
